.github/workflows/deploy.yml

```diff
@@ -28,36 +28,71 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
         with:
-          fetch-depth: 0 
+          fetch-depth: 0
       - uses: pnpm/action-setup@v3
         with:
-          version: 9 
+          version: 9
       - name: Setup Node
         uses: actions/setup-node@v4
         with:
           node-version: 20
           cache: pnpm # or pnpm / yarn
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.pnpm-store
+          key: ${{ runner.os }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-
       - name: Setup Pages
         uses: actions/configure-pages@v4
       - name: Install dependencies
         run: pnpm install
-      - name: Analyze 
-        run: pnpm analyze 
+      - name: Analyze
+        run: pnpm analyze
       - name: Build Docs
         run: pnpm docs:build
       - name: Upload artifact
         uses: actions/upload-pages-artifact@v3
         with:
           path: docs/.vitepress/dist
 
+  test:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
```

Comment:
There are some duplicated lines in the `build` and `test` stages, Please check if it can be factored